middleware/src/middleware.go

```diff
@@ -251,6 +256,74 @@ func (middleware *Middleware) Restore(method rpcmessages.RestoreArgs) (rpcmessag
 	}
 }
 
+// Hostname returns a GenericResponse struct in response to a rpcserver request
+func (middleware *Middleware) Hostname(args rpcmessages.HostnameArgs) (rpcmessages.GenericResponse, error) {
+	switch args.Method {
+	case rpcmessages.GetHostname:
+		log.Println("Getting the hostname via the config script")
+		out, err := middleware.runBBBConfigScript("get", "hostname", "")
+		if err != nil {
+			return rpcmessages.GenericResponse{Success: false, Message: string(out)}, err
+		}
+		return rpcmessages.GenericResponse{Success: true, Message: string(out)}, nil
+
+	case rpcmessages.SetHostname:
+		log.Println("Setting the hostname via the config script")
+		var r = regexp.MustCompile(`^[a-z0-9]+[a-z0-9-.]{0,62}[a-z0-9]$`)
+
+		if len(args.Hostname) >= 2 && len(args.Hostname) <= 64 && r.MatchString(args.Hostname) {
+			out, err := middleware.runBBBConfigScript("set", "hostname", args.Hostname)
+			if err != nil {
+				return rpcmessages.GenericResponse{Success: false, Message: string(out)}, err
+			}
+			return rpcmessages.GenericResponse{Success: true, Message: string(out)}, nil
+		}
+		errorMessage := fmt.Sprintf("invalid hostname '%s'", args.Hostname)
+		return rpcmessages.GenericResponse{Success: false, Message: errorMessage}, errors.New(errorMessage)
+
+	default:
+		errorMessage := fmt.Sprintf("Method %d not supported for Hostname().", args.Method)
+		return rpcmessages.GenericResponse{Success: false, Message: errorMessage}, errors.New(errorMessage)
+	}
+}
+
+// RootPassword returns a GenericResponse struct in response to a rpcserver request
+func (middleware *Middleware) RootPassword(args rpcmessages.RootPasswordArgs) (rpcmessages.GenericResponse, error) {
+	switch args.Method {
+	case rpcmessages.EnableRootLogin:
+		log.Println("Enabling the root login via the config script")
+		out, err := middleware.runBBBConfigScript("enable", "root_pwlogin", "")
+		if err != nil {
+			return rpcmessages.GenericResponse{Success: false, Message: string(out)}, err
+		}
+		return rpcmessages.GenericResponse{Success: true, Message: string(out)}, nil
+
+	case rpcmessages.DisableRootLogin:
+		log.Println("Disabling the root login via the config script")
+		out, err := middleware.runBBBConfigScript("disable", "root_pwlogin", "")
+		if err != nil {
+			return rpcmessages.GenericResponse{Success: false, Message: string(out)}, err
+		}
+		return rpcmessages.GenericResponse{Success: true, Message: string(out)}, nil
```

Comment:
Can you split this method up into a SetRootPassword method and a RootLoginPolicy method?